src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py

```diff
@@ -338,12 +334,12 @@ def __call__(
                 noise_pred = noise_pred_uncond + guidance_scale * (noise_pred_text - noise_pred_uncond)
 
             # compute the previous noisy sample x_t -> x_t-1
+            latents = self.scheduler.step(noise_pred, t, latents, **extra_step_kwargs).prev_sample
             if isinstance(self.scheduler, LMSDiscreteScheduler):
-                latents = self.scheduler.step(noise_pred, t_index, latents, **extra_step_kwargs).prev_sample
                 # masking
+                t_index = t_start + i
                 init_latents_proper = self.scheduler.add_noise(init_latents_orig, noise, torch.LongTensor([t_index]))
```

Comment:
Couldn't the `add_noise` function also swallow the complexity here by converting `t` to `t_index` ?

Reply:
`add_noise` expects indices instead of timesteps, which are in reverse order (same issue as img2img). Trying to find an elegant solution right now, open to suggestions

Reply:
Think the `add_noise` method should just be able handle the same input type as `add_noise` of the other schedulers